### PR TITLE
[RNMobile] Fix bug - multiple media selection adds only one image or video block

### PR DESCRIPTION
**Pull request by marecar3**
## Description
This PR addresses an [issue](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1346)
where multiple media selection adds only one image or video block on Android devices and on iOS where the Force Touch app shortcut doesn't work properly selecting "New Photo Post".

Related [gutenberg-mobile PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/1383): 

## How has this been tested?

**[Android]**

Current behavior:

1. Add an image/video block
2. Add multiple images/videos from WordPress media library
3. Only 1 image/video block is added

Expected behavior:
1. Add an image/video block
2. Add multiple images/videos from WordPress media library
3. Multiple image/video blocks are added with the selected images/videos

**[iOS]**

This also seems to be the cause of an issue on WPiOS, where the Force Touch app shortcut doesn't work properly selecting "New Photo Post".

Steps:

1. Force Touch on the WPiOS app icon.
2. Select New Photo Post shortcut.
3. Make sure Gutenberg was opened as the editor.
4. Select an Image.
5. 💥 No image block was appended.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

File: packages/editor/src/components/provider/index.native.js [hunk 1]
```diff
@@ -63,6 +64,16 @@ class NativeEditorProvider extends Component {
...
 		} );
+
+		this.subscriptionParentMediaAppend = subscribeMediaAppend( ( payload ) => {
+			const blockName = 'core/' + payload.mediaType;
+			const newBlock = createBlock( blockName, {
+				id: payload.mediaId,
+				[ payload.mediaType === 'image' ? 'url' : 'src' ]: payload.mediaUrl,
+			} );
+
+			this.props.insertBlock( newBlock, this.props.blockCount );
```
Comment (etoledom) [inline packages/editor/src/components/provider/index.native.js]: The only problem with this is that all images but the first one will be added at the end of the post.

Not sure if at this point we have enough information to add each new image when it corresponds? 🤔 
For sure can be an improvement for the future.

**Review by etoledom — COMMENTED** [review]
Thank you @marecar3 for taking care of this!

Tested on WPAndroid and WPiOS and this solves the issues in both platforms 🎉 

⚠️ There are some extra commits showing, probably it would be good to fix the history before merging.

**WPiOS:**
On iOS for some reason the resulting image is a very low quality one. This for sure is an issue on the Native side not related to this PR, but still wanted to highlight it.

ios-shortcut

**WPAndroid:**
On Android, there's a small inconvenience that, if I want to add the I mages at the beginning of the post (or anywhere that is not the end), the first image block is positioned correctly, but all the rest are added at the end of the block.

Not sure how important is this problem, but I'd me OK to make a ticket out of this to be handled in the future (cc @hypest @pinarol ):

add-multiple

**Comment by marecar3** [comment]
Hey @etoledom, thanks for the good review!

I fixed an issue when images are shown at the end of the post, not they will be shown bellow selected block which should be image/video block. Let me know what do you think about that approach. Thanks.

Regarding the low-quality image, I have experienced similar on Android but just managed to reproduce it once.

PR is ready for another iteration.

File: packages/editor/src/components/provider/index.native.js [hunk 2]
```diff
@@ -151,12 +166,21 @@ export default compose( [
...
+		const selectedBlockClientId = getSelectedBlockClientId();
+		const selectedBlockIndex = getBlockIndex( selectedBlockClientId );
+
 		return {
 			mode: getEditorMode(),
 			isReady: isEditorReady(),
 			blocks: getEditorBlocks(),
 			title: getEditedPostAttribute( 'title' ),
 			getEditedPostContent,
+			selectedBlockIndex: selectedBlockIndex,
```
Comment (etoledom) [inline packages/editor/src/components/provider/index.native.js]: Here you can avoid repeating `selectedBlockIndex`

**Review by etoledom — COMMENTED** [review]
Thank you for the update @marecar3 !

I just caught an issue with this new strategy, while it does solve the problem previously described, now the images are appended to the post on the reversed order from the selection.

My guess is that since we are appending always at the selected block index, every next image is added always at the top index.

multiple_media

Again, I'm not sure if there's an easy way to make this work properly without digging into the code more deeply, and I don't recall what was the behavior previously. 🤔

File: packages/editor/src/components/provider/index.native.js (hunk 2 above)
Comment (marecar3) [reply packages/editor/src/components/provider/index.native.js]: Good catch! Fixed!

**Comment by marecar3** [comment]
Hey, @etoledom thanks for the good review!

I saw the old code and took the solution from here: https://github.com/wordpress-mobile/gutenberg-mobile/blob/cb93dd24e50855f15f43f2ce00b2eaa76c1bd8ef/src/block-management/block-manager.js#L115

I tested and didn't find any problems. Let me know your opinion on this one.

PR is ready for another iteration of testing. Thanks!!!

**Review by etoledom — APPROVED** [review]
Tested on both iOS and Android again.
iOS keep working as mentioned before (fixed the original issue 🎉 )
Android is not working as expected! Nice job @marecar3 🥇

**Comment by etoledom** [comment]
There are a couple of lint errors:
```
[0] /home/travis/build/WordPress/gutenberg/packages/editor/src/components/provider/index.native.js
835[0]   179:1   error  Trailing spaces not allowed  no-trailing-spaces
836[0]   187:45  error  Missing trailing comma       comma-dangle
```

The autosave tests error (as far as I understand) are safe to ignore for now.

**Comment by marecar3** [comment]
Hey @etoledom, thanks for the approve and for good review 👍 

I have fixed the lint tests and updated release notes. 
When tests pass I will merge this PR.

**Pull request merged**